pkg/storaged/logs-panel.jsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { journalctl, renderer } from "journal";
+
+const _ = cockpit.gettext;
+
+// XXX - turn this into something reusable
+
+class JournalOutput {
+    constructor() {
+        this.logs = [ ];
+    }
+
+    render_line(ident, prio, message, count, time, entry) {
```

Comment:
I don't see this function being called anywhere? Instead you import and call `render()` from `journal.js` which I believe is pkg/lib/journal.js (the moustache one).

Reply:
Yes, this needs comments.  `JournalOutput` implements the API expected by `journal.renderer`.